### PR TITLE
Make sure ObstructedMaze is solvable.

### PR DESCRIPTION
**Pull request by SigmaBM**
# Description

The blocking ball placed by the `add_door` may cover the box placed on the map by previous `add_door`, causing the agent to be unable to open certain door and unable to complete the task. To make sure this environment solvable, we can first place all the doors and their blocking balls, and then place the keys, so that it ensures that the keys will not be covered by the blocking balls added later.

Fixes # 323

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### Screenshots
Please attach before and after screenshots of the change if applicable.

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]
Could you
1. Provide a test to show the number of environments that are not solvable out of 10,000? 
2. update the version numbers for all Obstructed Maze
3. Add a test to check if the environment is solvable and run 10 times in the tests.

**Comment by SigmaBM** [comment]
Could you please clarify if "update the version numbers" means changing from "v0" to "v1"? Also, which directory should I place the test files? Thank you.

**Comment by pseudo-rnd-thoughts** [comment]
> Could you please clarify if "update the version numbers" means changing from "v0" to "v1"? Also, which directory should I place the test files? Thank you.

Yes, just updated the registration version number, see `minigrid/__init__.py` I believe and also update the documentation for the changes from v0 to v1

For testing, could you put it in `tests/test_obstructed_maze.py`

**Comment by SigmaBM** [comment]
I have updated the version number of related ObstructedMaze envs and the document. A test file is available in `tests/test_obstructed_maze.py`.

File: minigrid/envs/obstructedmaze.py [hunk 1]
```diff
@@ -58,16 +58,21 @@ class ObstructedMazeEnv(RoomGrid):
     "Q" number of quarters that will have doors and keys out of the 9 that the
     map already has.
     "Full" 3x3 maze with "h" and "b" options.
+    "v1" prevent the key from being covered by the blocking ball.
```
Comment (pseudo-rnd-thoughts) [inline minigrid/envs/obstructedmaze.py:61]: Could you update to explain why only some of the environments are updated

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]
Could you add a test, `test_solvable_env` that the `env_test` for the v1 envs have a unsolvable == 0

**Review by pseudo-rnd-thoughts — APPROVED** [review]
Thanks, I think this is good.

**Pull request merged**